gpytorch/lazy/interpolated_lazy_variable.py

```diff
@@ -0,0 +1,280 @@
+                                                                self.base_lazy_variable.size()[-1])
+            right_factor = right_factor.transpose(-1, -2)
+            right_res = bdsmm(right_interp, right_factor).transpose(-1, -2).contiguous()
+
+            res = tuple(list(base_lazy_var_deriv(left_res, right_res)) + [None] * 4)
+            return res
+
+        return closure
+
+    def _transpose_nonbatch(self):
+        res = self.__class__(self.base_lazy_variable.transpose(-1, -2), self.right_interp_indices,
+                             self.right_interp_values,
+                             self.left_interp_indices, self.left_interp_values)
+        return res
+
+    def diag(self):
+        n_data, n_interp = self.left_interp_indices.size()
+
+        # Batch compute the non-zero values of the outer products w_left^k w_right^k^T
+        left_interp_values = self.left_interp_values.unsqueeze(2)
+        right_interp_values = self.right_interp_values.unsqueeze(1)
+        interp_values = torch.matmul(left_interp_values, right_interp_values)
+
+        # Batch compute Toeplitz values that will be non-zero for row k
+        left_interp_indices = self.left_interp_indices.unsqueeze(2).expand(n_data, n_interp, n_interp).contiguous()
+        right_interp_indices = self.right_interp_indices.unsqueeze(1).expand(n_data, n_interp, n_interp).contiguous()
+        base_var_vals = self.base_lazy_variable._get_indices(left_interp_indices.view(-1),
+                                                             right_interp_indices.view(-1))
+        base_var_vals = base_var_vals.view(left_interp_indices.size())
+
+        diag = (interp_values * base_var_vals).sum(2).sum(1)
+        return diag
+
+    def mul(self, other):
+        """
+        Multiplies the matrix by a constant, or elementwise the matrix by another matrix
+        """
+        if not (isinstance(other, Variable) or isinstance(other, LazyVariable)) or \
+               (isinstance(other, Variable) and other.numel() == 1):
+            base_lazy_variable = ConstantMulLazyVariable(self.base_lazy_variable, other)
+
+            return self.__class__(base_lazy_variable, self.left_interp_indices, self.left_interp_values,
+                                  self.right_interp_indices, self.right_interp_values)
+
+        else:
+            from .mul_lazy_variable import MulLazyVariable
+            return MulLazyVariable(self, other)
+
+    def repeat(self, *sizes):
+        """
+        Repeat elements of the Variable.
+        Right now it only works to create a batched version of a ToeplitzLazyVariable.
+
+        e.g. `var.repeat(3, 1, 1)` creates a batched version of length 3
+        """
+        if not len(sizes) == 3 and sizes[1] == 1 and sizes[2] == 1:
+            raise RuntimeError('Repeat only works to create a batched version at the moment.')
+        if self.J_left is None:
+            raise RuntimeError('Repeat only works when the toeplitz variable is interpolated.')
```

Comment:
This probably isn't a great error message now that this lives in `InterpolatedLazyVariable`

Reply:
Will fix.